lib/utils/github.ts

```diff
@@ -86,10 +86,52 @@ const getPullRequestCommitMessageFromUrl = async (url: string): Promise<string[]
   return (data as Commit[]).map((commit: Commit): string => commit.commit.message);
 };
 
+const getGithubIssueDetails = async (url: string): Promise<any> => {
+  const sessionResponse = await supabase.auth.getSession();
+  const githubToken = sessionResponse?.data.session?.provider_token;
+  const [, , , owner, repoName, , issueNumber] = url.split("/");
+
+  const apiUrl = `https://api.github.com/repos/${owner}/${repoName}/issues/${issueNumber}`;
+
+  const response = await fetch(apiUrl, {
+    headers: {
+      Authorization: `token ${githubToken}`,
+    },
+  });
+  const data = await response.json();
+
+  return {
+    title: data.title,
+    body: data.body,
+  };
+};
+
+const getGithubIssueComments = async (url: string): Promise<any> => {
+  const sessionResponse = await supabase.auth.getSession();
+  const githubToken = sessionResponse?.data.session?.provider_token;
+  const [, , , owner, repoName, , issueNumber] = url.split("/");
+
+  const apiUrl = `https://api.github.com/repos/${owner}/${repoName}/issues/${issueNumber}/comments`;
+
+  const response = await fetch(apiUrl, {
+    headers: {
+      Authorization: `token ${githubToken}`,
```

Comment:
If we want to go down the octokit path, we can use it's hydration methods to get auth into our octokit client:
https://octokit.github.io/rest.js/v20#authentication

```js
const { Octokit } = require("@octokit/rest");

const octokit = new Octokit({
  auth: "mypersonalaccesstoken123",
});
```